### PR TITLE
DSL: Add converters for string field

### PR DESCRIPTION
**Pull request by hfazai**

### CODE REVIEW

File: galite-core/src/main/kotlin/org/kopi/galite/domain/Domain.kt [hunk 1]
```diff
@@ -72,12 +74,25 @@ open class Domain<T>(val width: Int? = null,
...
-                                      height ?: 1,
-                                      visibleHeight ?: 1,
-                                      0,  // TODO
-                                      styled)
+        String::class -> {
+          if(visibleHeight != height) {
+            VStringField(block.buffer,
+                         width ?: 0,
+                         height ?: 1,
+                         visibleHeight ?: 0,
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/domain/Domain.kt:82]: can you check here if we need to put **?: 0** or **?: 1**

File: galite-core/src/main/kotlin/org/kopi/galite/domain/Domain.kt [hunk 2]
```diff
@@ -72,12 +74,25 @@ open class Domain<T>(val width: Int? = null,
...
+                         styled)
+          } else {
+            VStringField(block.buffer,
+                         width ?: 0,
+                         height ?: 1,
+                         (fixed.value or convert.value) and
+                                 VConstants.FDO_CONVERT_MASK or
+                                 VConstants.FDO_DYNAMIC_NL,
+                         styled)
+          }
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/domain/Domain.kt:94]: can we replace this bloc by :
```
VStringField(block.buffer,
                         width ?: 0,
                         height ?: 1,
                         if(visibleHeight != height) visibleHeight ?: 0 else 0,
                         (fixed.value or convert.value) and
                                 (VConstants.FDO_CONVERT_MASK or VConstants.FDO_DYNAMIC_NL),
                         styled)
```

**Review by h-haddad — APPROVED** [review]

File: galite-core/src/main/kotlin/org/kopi/galite/domain/Domain.kt (hunk 1 above)
Comment (hfazai) [reply galite-core/src/main/kotlin/org/kopi/galite/domain/Domain.kt:82]: It should be 0 if `visibleHeight` is `null`

File: galite-core/src/main/kotlin/org/kopi/galite/domain/Domain.kt (hunk 2 above)
Comment (hfazai) [reply galite-core/src/main/kotlin/org/kopi/galite/domain/Domain.kt:94]: They are two different constructors

**Pull request merged**